### ISSUE TITLE
Problem with TypeTestClosure

### ISSUE THREAD

**Issue by DavePearce** [Whiley/WhileyTheoremProver#65]
The following verifies when it clearly shouldn't:

<pre>
type nat is (int x) where x >= 0
type pos is (int x) where x > 0

assert:
    forall (int x):
       if:
          x is nat
       then:
          x is pos  
</pre>

The key problem is that `TypeTestClosure` reduces this to `x is (nat&!pos)` (effectively `x is (int&!int)`) which clearly gives a contradiction.  But, of course, this is ignoring the constraints involved.

The problem essentially boils down to this segment:

<pre>
Type intersection = new Type.Intersection(lhsT,rhsT);
	//
	if (types.isRawSubtype(new Type.Void(), intersection)) {
		// No possible intersection exists between the types in
		// question. Therefore, the test cannot be true.
		return state.subsume(this, typeTest, state.allocate(new Formula.Truth(false)));
	} 
</pre>

Basically, it's not true in the case of constrained types that `isRawSubtype` determines categorically that there is no intersection between the two types.  For example, consider `nat :> pos` which corresponds to `int :> int` as a raw subtype test (which is clearly incorrect).

**How do we handle this properly?** The test `nat :> pos` translates to the void test `(!nat) & pos`.  To determine the type really is equivalent to void, we want to compute <code>&lceil;(!nat)&pos&rceil;</code> which expands to <code>(!&lfloor;nat&rfloor;)&&lceil;pos&rceil;</code> and then `(!void)&int` (which does not return true).
